lib/constructs/ApiSfnFunction.ts

```diff
@@ -0,0 +1,80 @@
+import { Duration } from "aws-cdk-lib";
+import * as lambda from "aws-cdk-lib/aws-lambda";
+import * as lambdaNodeJs from "aws-cdk-lib/aws-lambda-nodejs";
+import * as sfn from "aws-cdk-lib/aws-stepfunctions";
+import * as iam from "aws-cdk-lib/aws-iam";
+import { HttpMethod } from "../../utils/http";
+import { Construct } from "constructs";
+
+/**
+ * An IAM service principal for the API Gateway service, used to grant Lambda
+ * invocation permissions.
+ */
+const APIGW_SERVICE_PRINCIPAL = new iam.ServicePrincipal("apigateway.amazonaws.com");
+
+export interface ApiSfnFunctionProps {
+  /**
+   * The HTTP method this route applies to.
+   */
+  readonly method: HttpMethod;
+
+  /**
+   * The local path where the handler file is located.
+   */
+  readonly handlerPath: string;
+
+  /**
+   * Any additional props to apply to the NodeJsFunction.
+   *
+   * Additional prop values specified here will override the defaults, including
+   * the `entry` and `environment` if one is provided in this struct. Be careful.
+   */
+  readonly functionPropsOverride?: lambdaNodeJs.NodejsFunctionProps;
+
+  /**
+   * The State Machine that will start execution when this function
+   * is called.
+   */
+  readonly stateMachine: sfn.IStateMachine;
+}
+
+export class ApiSfnFunction extends Construct {
```

Comment:
We ran into a lot of complexity with creating the `ApiXFunction`s before... should we talk about whether this is still the way we want to go?

Reply:
To be honest, I was going down the path of trying to implement the function using the decorator pattern (as previously suggested), but it was an epic fail and I wasn't getting anywhere. Hence, I did this for the time being to ensure functionality first.